src/app/header/components/SearchBar.tsx

```diff
@@ -1,29 +1,49 @@
 import { IconButton, Input, InputGroup, InputRightAddon } from '@chakra-ui/react';
-import React, { useState } from 'react';
+import React, { ChangeEvent } from 'react';
 import { GrSearch } from 'react-icons/gr';
+import { SearchBoxProvided } from 'react-instantsearch-core';
+import { connectSearchBox } from 'react-instantsearch-dom';
 
-import { useChangeCallback } from '../hooks/useChangeCallback';
+type SearchBoxProps = {
+  currentRefinement: string;
+  isSearchStalled: boolean;
+  refine: (value: string) => void;
+  onChange?: (query: string) => void;
+  onSubmit?: (query: string) => void;
+};
 
-export function SearchBar(): JSX.Element {
-  const [value, setValue] = useState('');
-  const onChange = useChangeCallback(setValue);
+function SearchBox({ currentRefinement, isSearchStalled, refine, onChange, onSubmit }: SearchBoxProps) {
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    refine(e.currentTarget.value);
+    onChange && onChange(e.currentTarget.value);
+  };
 
-  //TODO: add search functionality
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    onSubmit !== undefined && onSubmit(currentRefinement);
+  };
 
   return (
-    <InputGroup size="sm">
-      <Input
-        placeholder="Search a course"
-        isFullWidth
-        bg="white"
-        w="100%"
-        minW="100px"
-        value={value}
-        onChange={onChange}
-      />
-      <InputRightAddon p="0">
-        <IconButton aria-label="Search" borderRadius="0" size="sm" icon={<GrSearch />} />
-      </InputRightAddon>
-    </InputGroup>
+    <form noValidate action="" role="search" onSubmit={handleSubmit}>
+      <InputGroup size="sm">
+        <Input
+          placeholder="Search for courses"
+          bg="white"
+          width={['xs', 'sm', 'md']}
+          value={currentRefinement}
+          onChange={handleChange}
+        />
+        <InputRightAddon p="0">
+          <IconButton aria-label="Search" borderRadius="0" size="sm" icon={<GrSearch />} />
```

Comment:
this can probably be removed now, think it's redundant

Reply:
The icon?

Reply:
Yeah, since pressing it doesn't do anything